src/librustc/ty/error.rs

```diff
@@ -246,6 +246,37 @@ impl<'tcx> ty::TyS<'tcx> {
             ty::Error => "type error".into(),
         }
     }
+
+    pub fn prefix_string(&self) -> Cow<'static, str> {
+        debug!("prefix_string {:?} {} {:?}", self, self, self.kind);
+        match self.kind {
+            ty::Infer(_) | ty::Error | ty::Bool | ty::Char | ty::Int(_) |
+            ty::Uint(_) | ty::Float(_) | ty::Str | ty::Never => "type".into(),
+            ty::Tuple(ref tys) if tys.is_empty() => "type".into(),
```

Comment:
```suggestion
            ty::Tuple(ref tys) if tys.is_empty() => "unit type".into(),
```

Reply:
changing this will result in

    expected unit type `()`

Is that what we'd like?